lib/compat/wordpress-5.9/block-template-utils.php

```diff
@@ -37,7 +37,11 @@ function get_block_theme_folders( $theme_stylesheet = null ) {
 		$root_dir   = get_theme_root( $theme_name );
 		$theme_dir  = "$root_dir/$theme_name";
 
-		if ( is_readable( $theme_dir . '/block-templates/index.html' ) ) {
+		$parent_theme_name = get_template();
+		$parent_root_dir   = get_theme_root( $parent_theme_name );
+		$parent_theme_dir  = "$parent_root_dir/$parent_theme_name";
+
+		if ( is_readable( $theme_dir . '/block-templates/index.html' ) || is_readable( $parent_theme_dir . '/block-templates/index.html' ) ) {
```

Comment:
The function here is supposed to be called with a theme stylesheet, so there shouldn't be a need for this added check. Maybe there's an issue elsewhere where we call this and we're not calling it with the parent theme as well?

Reply:
Forget the first comment, I understand the issue better now and I have separate thoughts :) 

What happens if the child theme has been migrated but not the parent, it seems this PR will break that behavior.

Reply:
Maybe the fix is to actually check the presence of the folder and not the index file.

Reply:
Yeah that would work I think, although I guess its possible that a child theme could not have the folder at all and just supply a theme.json.

If the child has been migrated before the parent then I would expect things to break to be honest!

Reply:
> Yeah that would work I think, although I guess its possible that a child theme could not have the folder at all and just supply a theme.json.

In that case, it doesn't matter since this function is about returning the folders.

Reply:
Good point :)